p2p-tests/maintainer-update-propagation-1-test.ts

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env -S node --require ts-node/register/transpile-only --require tsconfig-paths/register
+
+// Copyright © 2021 The Radicle Upstream Contributors
+//
+// This file is part of radicle-upstream, distributed under the GPLv3
+// with Radicle Linking Exception. For full terms see the included
+// LICENSE file.
+
+import * as path from "path";
+import { strict as strictAssert } from "assert";
+
+import * as source from "proxy-client/source";
+import { sleep } from "ui/src/sleep";
+import {
+  UpstreamPeer,
+  UpstreamSeed,
+  commit,
+  pushRad,
+  runTestcase,
+  withRetry,
+} from "./lib/p2p";
+
+let maintainer: UpstreamPeer;
+let contributor: UpstreamPeer;
+let seed: UpstreamSeed;
+
+async function testcase(dataPath: string) {
+  maintainer = new UpstreamPeer({
+    namespace: "maintainer",
+    ipAddress: "10.0.0.101",
+    startArgs: [
+      "--seed",
+      "hybfoqx9wrdjhnr9jyb74zpduph57z99f67bjgfnsf83p1rk7z1diy@10.0.0.1:8776",
+    ],
+    dataPath,
```

Comment:
The `--rad-home` option and listen options passed to the peer are represented as fields while the `--seed` option is passed via the generic `startArgs` parameter. I think it would make sense to have all options be provided as parameters and have the internals of the class translate them.

```suggestion
    ipAddress: "10.0.0.101",
    seed: "hybfoqx9wrdjhnr9jyb74zpduph57z99f67bjgfnsf83p1rk7z1diy@10.0.0.1:8776",
    dataPath,
```

Reply:
e2e745a183320591a7fbfdb8bf86ae3e8d4a1fa9